library-app/handlers/author_handler.go

```diff
@@ -0,0 +1,206 @@
+package handlers
+
+import (
+	"awesomeProject/library-app/global"
+	"awesomeProject/library-app/managers/interfaces"
+	"awesomeProject/library-app/models"
+	"encoding/json"
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	log "github.com/sirupsen/logrus"
+	"net/http"
+)
+
+type AuthorHandler struct {
+	authorManager interfaces.AuthorManager
+}
+
+func NewAuthorHandler(authorManager interfaces.AuthorManager) *AuthorHandler {
+	return &AuthorHandler{authorManager}
+}
+
+func (h *AuthorHandler) GetAll(w http.ResponseWriter, r *http.Request) {
+	log.Info("[AuthorHandler.GetAll] Fetching all authors")
+
+	authors, dbErr := h.authorManager.GetAll()
+	if dbErr != nil {
+		global.HttpDBError(
+			w,
+			dbErr,
+		)
+		return
+	}
+
+	err := json.NewEncoder(w).Encode(authors)
+	if err != nil {
+		global.HttpError(
+			w,
+			"[AuthorHandler.GetAll] Failed to encode authors to JSON",
+			"Failed to return authors",
+			http.StatusInternalServerError,
+			err,
+		)
+		return
+	}
+}
+
+func (h *AuthorHandler) Get(w http.ResponseWriter, r *http.Request) {
+	log.Info("[AuthorHandler.Get] Fetching author")
+
+	vars := mux.Vars(r)
+
+	id, err := uuid.Parse(vars["id"])
+	if err != nil {
+		global.HttpError(
+			w,
+			"[AuthorHandler.Get] Invalid UUID format",
+			"Invalid author ID format",
+			http.StatusBadRequest,
+			err,
+		)
+		return
+	}
+
+	author, dbErr := h.authorManager.Get(id)
+	if dbErr != nil {
+		global.HttpDBError(
+			w,
+			dbErr,
+		)
+		return
+	}
+
+	err = json.NewEncoder(w).Encode(author)
+	if err != nil {
+		global.HttpError(
+			w,
+			"[AuthorHandler.Get] Failed to encode author to JSON",
+			"Failed to return author",
+			http.StatusInternalServerError,
+			err,
+		)
+	}
+}
+
+func (h *AuthorHandler) Create(w http.ResponseWriter, r *http.Request) {
+	log.Info("[AuthorHandler.Create] Creating new author")
+
+	var newAuthor models.Author
+	err := json.NewDecoder(r.Body).Decode(&newAuthor)
+	if err != nil {
+		global.HttpError(
+			w,
+			"[AuthorHandler.Create] Failed to decode JSON body into Author struct",
+			"Invalid JSON format in request body",
+			http.StatusBadRequest,
+			err,
+		)
+		return
+	}
+
+	newAuthor.ID = uuid.Nil
+	createdAuthor, dbErr := h.authorManager.Create(newAuthor)
+	if dbErr != nil {
+		global.HttpDBError(
+			w,
+			dbErr,
+		)
+		return
+	}
+
+	err = json.NewEncoder(w).Encode(createdAuthor)
+	if err != nil {
+		global.HttpError(w,
+			"[AuthorHandler.Create] Failed to encode created author to JSON",
+			"Failed to return created author",
+			http.StatusInternalServerError,
+			err,
+		)
+	}
+}
+
+func (h *AuthorHandler) Update(w http.ResponseWriter, r *http.Request) {
+	log.Info("[AuthorHandler.Update] Updating author")
+
+	vars := mux.Vars(r)
+
+	id, err := uuid.Parse(vars["id"])
+	if err != nil {
+		global.HttpError(w,
+			"[AuthorHandler.Update] Invalid UUID format",
+			"Invalid author ID format",
+			http.StatusBadRequest,
+			err,
+		)
+		return
+	}
+
+	var updatedAuthorBody models.Author
```

Comment:
With this pattern you expect the ID of object to be present in path and in payload, that is redundant. 
More useful approach is to expect ID in path only and to load object before updating it.
With this we increase usability by affected performance.

Reply:
After this I'm setting updatedAuthorBody.ID = id, where id is the id from the path, so i don't expect the id to be in the payload.